lib/sudo-system/src/lib.rs

```diff
@@ -140,11 +140,14 @@ pub fn set_target_user(
     }
 }
 
-/// Send a signal to a process.
-pub fn kill(pid: ProcessId, signal: c_int) -> io::Result<()> {
+/// Send a signal to a process with the specified ID.
+///
+/// if the ID is `None`, then the signal is sent to every process for which the calling process has
+/// permission to send signal except for the init process.
+pub fn kill(pid: impl Into<Option<ProcessId>>, signal: c_int) -> io::Result<()> {
```

Comment:
is there any caller of this function that passes `None` as the first argument? if not, I'd prefer to change the first parameter to `ProcessId` and leave out the `.unwrap_or(-1)` from the implementation.